kiliautoml/models/_pytorchvision_image_classification.py

```diff
@@ -194,27 +211,56 @@ def find_errors(
     ) -> Any:
         _ = clear_dataset_cache
 
-        original_image_datasets, labels, class_names = self.prepare_dataset(assets, api_key)
+        images = download_project_images(
+            api_key=api_key, assets=assets, output_folder=self.data_dir
+        )
+        labels = []
+        for asset in assets:
+            label = get_label(asset, label_merge_strategy)
+            if (label is None) or (self.job_name not in label["jsonResponse"]):
+                asset_id = asset["id"]
+                warnings.warn(f"${asset_id}: No annotation for job ${self.job_name}")
+                return 0.0
+            else:
+                labels.append(label["jsonResponse"][self.job_name]["categories"][0]["name"])
+
         kf = StratifiedKFold(n_splits=cv_n_folds, shuffle=True, random_state=42)
+        pyx = np.empty((len(labels), len(self.class_name_to_idx)))
+
         for cv_fold in tqdm(range(cv_n_folds), desc="Training and predicting on several folds"):
             # Split train into train and holdout for particular cv_fold.
             cv_train_idx, cv_holdout_idx = list(kf.split(range(len(labels)), labels))[cv_fold]
-
-            image_datasets, holdout_dataset = separe_holdout_datasets(
-                cv_n_folds,
-                verbose,
-                original_image_datasets,
-                cv_fold,
-                cv_train_idx,
-                cv_holdout_idx,
+            splits = {}
+            splits["train"], splits["val"] = train_test_split(
+                cv_train_idx, test_size=0.2, random_state=42
             )
+            image_datasets = {
+                x: ClassificationTrainDataset(
```

Comment:
does it make sense to rename to  ClassificationDataset ?

Reply:
It is to differentiate between ClassificationTrainDataset and ClassificationPredictDataset where one store labels, the other not. But I am open to a renaiming